### DISCUSSION TITLE
Is it possible to directly stream the data from Azure Blob Store

### DISCUSSION THREAD

**Discussion by rishabhgupta93** [trhallam/segysak#128] [Q&A] [answered]
Hi,

I want to stream the segy file directly from azure blob store. is there a way to directly stream the data and avoid copying of data in local VM ?

Thanks

**Comment by trhallam** [comment, marked as answer]
Afraid not, SEG-Y data was designed at a time when streaming data wasn't even an idea in someones mind. If you want to stream data, you should convert the SEG-Y to a file format that supports brick/block streaming with small meta-data overheads like OpenVDS, Zarr or perhaps ZGY (theoretically possible, but I'm not sure how mature the library is to support it). SEGY-SAK can help you with the conversion, but the data will need to be accessible on a local file system first.
